tests/e2e/utils/response_evaluation.py

```diff
@@ -143,3 +151,68 @@ def validate_response(self):
             print("No result. Nothing to process.")
 
         return True
+
+    def evaluate_models(self):
+        """Evaluate models against groundtruth answer."""
+        print("Running model evaluation using groundtruth...")
+        inscope_models = [
+            "bam+ibm/granite-13b-chat-v2",
+            "watsonx+ibm/granite-13b-chat-v2",
+            "openai+gpt-3.5-turbo",
+            "azure_openai+gpt-3.5-turbo",
+        ]
+        answer_id = "ground_truth+with_rag"
+
+        result_df = DataFrame(columns=["eval_id", "question", answer_id])
+        for provider_model in inscope_models:
+            result_dict = defaultdict(list)
+            print(f"Model evaluation for {provider_model}")
+            provider, model = provider_model.split("+")
+            try:
+                prev_resp = read_csv(
+                    f"{self._args.eval_out_dir}/response_evaluation_result-"
+                    f"{provider_model.replace('/', '-')}+with_rag.csv"
+                )
+            except FileNotFoundError:
+                prev_resp = None
+
+            for query_id in self._qa_pairs.keys():
+                question = self._qa_pairs[query_id]["question"]
+                answer_data = self._qa_pairs[query_id]["answer"][answer_id]
+                answer = answer_data["text"][0]
+
+                if prev_resp is not None:
+                    try:
+                        response = prev_resp[prev_resp.eval_id == query_id][
+                            "llm_response"
+                        ].iloc[0]
+                    except IndexError:
+                        response = self._get_api_response(question, provider, model)
+                else:
+                    response = self._get_api_response(question, provider, model)
+                score = self._similarity_score(response, answer)
+
+                result_dict["eval_id"].append(query_id)
+                result_dict["question"].append(question)
+                result_dict[answer_id].append(answer)
+                result_dict[f"{provider_model}_response"].append(response)
+                result_dict[f"{provider_model}_score"].append(score)
+
+            model_result_df = DataFrame.from_dict(result_dict)
+            result_df = result_df.merge(
+                model_result_df, on=["eval_id", "question", answer_id], how="outer"
+            )
+
+        result_df.to_csv(
+            f"{self._args.eval_out_dir}/model_evaluation_result.csv", index=False
+        )
+        summary_dict = {}
+        for m in inscope_models:
+            summary_dict[m] = result_df[f"{m}_score"].mean()
+        summary_dict = {
+            "timestamp": str(datetime.now(UTC)),
+            "eval_set": result_df.eval_id.unique().tolist(),
+            "avg_similarity_score": summary_dict,
+        }
+        with open(f"{self._args.eval_out_dir}/model_evaluation_summary.json", "w") as f:
+            json.dump(summary_dict, f)
```

Comment:
prob. I'm lost, but does it mean that every test that call this function, will spit some JSONs?